fe/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -542,15 +542,92 @@ private static DateTimeFormatterBuilder formatBuilder(String pattern) throws Ana
         return builder;
     }
 
-    public DateLiteral plusDays(int day) throws AnalysisException {
+    public DateLiteral plusYears(int year) throws AnalysisException {
+        LocalDateTime dateTime;
+        if (type == Type.DATE) {
+            dateTime = DATE_FORMATTER.parseLocalDateTime(getStringValue()).plusYears(year);
+        } else if (type == Type.DATETIME){
+            dateTime = DATE_TIME_FORMATTER.parseLocalDateTime(getStringValue()).plusYears(year);
+        } else {
+            throw new AnalysisException("Not support date literal type");
+        }
```

Comment:
It will reduce a lot of code if make these a function.
```
LocalDateTime dateTime = getTimeFormatter();
```

Reply:
Done